listeners/measured.go

```diff
@@ -46,11 +96,23 @@ func (c *wrapMeasuredConn) OnState(s http.ConnState) {
 // Responds to the "measured" message type
 func (c *wrapMeasuredConn) ControlMessage(msgType string, data interface{}) {
 	if msgType == "measured" {
-		c.Conn.ID = data.(string)
+		ctxUpdate := data.(map[string]interface{})
+		c.ctxMx.Lock()
+		defer c.ctxMx.Unlock()
+		newContext := make(map[string]interface{}, len(c.ctx))
+		// Copy context
+		for key, value := range c.ctx {
+			newContext[key] = value
+		}
+		// Update context
+		for key, value := range ctxUpdate {
+			newContext[key] = value
+		}
+		c.ctx = newContext
```

Comment:
I went ahead and moved the Unlock out of the defer so that we at least unlock prior to passing the control message along. We do however need to lock this entire block because we don't want concurrent updates trying to merge into the context at the same time.